styles/transition_pulse.h

```diff
@@ -0,0 +1,53 @@
+#ifndef STYLES_TRANSITION_PULSE_H
+#define STYLES_TRANSITION_PULSE_H
+
+#include "../transitions/concat.h"
+#include "../functions/effect_increment.h"
+
+// Usage: TransitionPulseL<TRANSITION, PULSE>
+// TRANSITION: TRANSITION
+// PULSE: FUNCTION
+// return value: COLOR
+//
+// When the specified PULSE happens TRANSITION layer will run.
+
+template<class TRANSITION, class PULSE>
+class TransitionPulseL {
+public:
+  LayerRunResult run(BladeBase* blade) {
+    pulse_.run(blade);
+    if (pulse_.calculate(blade)) {
+      transition_.begin();
+      run_ = true;
+    }
+    if (run_) {
+      transition_.run(blade);
+      if (transition_.done()) run_ = false;
+    }
+    last_detected_blade_effect = nullptr;
+    if (!run_) {
+      // All effects now wake the blade up again if needed, so it's safe to
+      // always return TRANSPARENT_UNTIL_IGNITION
+      return LayerRunResult::TRANSPARENT_UNTIL_IGNITION;
```

Comment:
This return (and the if statement) needs to go away, because it's not valid for pulses.